packages/coverage-istanbul/src/provider.ts

```diff
@@ -105,6 +105,11 @@ export class IstanbulCoverageProvider implements CoverageProvider {
       return map
     }, {})
 
+    if (!this.ctx.isFirstRun && !this.ctx.config.coverage.watchAll && this.ctx.changedTests.size > 0) {
+      const changedTests = Array.from(this.ctx.changedTests)
+      mergedCoverage.filter(key => key === changedTests[0].replace('.test', ''))
```

Comment:
This assumes that the tested file only imports files with similar names, right? So after you make change to `math.test.ts`, you will only see `math.ts`in report. You won't see `random-number-generator.ts` that it might import as well.

Also making changes to `math.spec.ts` breaks this logic as the hard-coded `.test` pattern is used.

Reply:
Yes, this is correct.